### PR TITLE
Added support for rescheduling tasks in periods of months and years.

### LINKED ISSUE #5: RFC: Support time periods other than days (months, years)
I use a lot of tasks that recur once a month, once every three months, once a year, etc. The underlying date implementations support syntax for other time periods, like months and years.

GNU implementation
$ date -d "2015-08-29 3 months" +%Y-%m-%d
2015-11-29

BSD implementation
$ date -j -v3m -f %Y-%m-%d 2015-08-29 +%Y-%m-%d
2015-11-29 (untested; according to documentation)

I think it would make sense to add support for this to again.

**Why not use just use the number of days in a month/year?**
Some recurring tasks, like "pay rent," are always due on the same day of the month. Since months have different numbers of days, I'd have to think too hard each month about whether to add 28 (February), 29 (leap year February), 30 (some months), or 31 (most months). I'd rather just type "todo.sh again N 1m" to increment the month but leave the day alone.

**Problem with late dates**
What should be the result of (2015-08-31 + 1 month)? 2015-09-31 is not a valid date. BSD will decrement the day number and return 2015-09-30. Over time, this will result in later days (29, 30, 31) drifting to 30 (short months) and eventually to 28 (February). GNU will always increment by 31 days if the day number would not be valid in the next month, which in this case returns 2015-10-01.

I personally like the BSD behavior better; I don't mind as much if my dates are early by a couple of days, but missing a task deadline is bad. So, again might have to build in some extra smarts to provide the BSD behavior (using awk/sed?) even when the GNU implementation is present.

### PR DESCRIPTION
**Pull request by jbrc**
Fixes #5 

New unitTests.sh and additions to testAgain.sh work on my GNU system. These changes are big enough that I'd like to test on a BSD system before you accept, but I thought I would go ahead and post to see if you have any feedback. I'll add more notes once I've tested the new code with a BSD system.

### CODE REVIEW

File: againHelpers.sh [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+#
+#You should have received a copy of the GNU General Public License
+#along with again.  If not, see <http://www.gnu.org/licenses/>.
+
+readonly DATE_FORMAT="%Y-%m-%d"
+
+# Adjust $ORIGINAL by $ADJUST, store in $NEW_DATE
+function adjust_date()
+{
+  ADJUST_NUM=`expr match "$ADJUST" '+\?\([0-9]\+\)'`
```
Comment (jbrc) [inline againHelpers.sh]: expr match isn't supported on BSD according to the man page ("these arguments are treated just as their respective string values", FreeBSD 10.2).

**Comment by nthorne** [comment]
This looks really good! Have you been able to give it a spin on a BSD system? If not, then I might be able to spin up a FreeBSD VM at the end of this weekend, or at the start of next week.

**Comment by jbrc** [comment]
Yep, I spent all day yesterday doing that on a FreeBSD 10.2 virtual machine. My update to this pull request (the 6th commit, 6463c97) includes fixes to problems uncovered by that testing.

So, I've tested on FreeBSD 10.2, and on a Windows 7 system running cygwin. unitTests.sh and testAgain.sh report all tests pass on both systems.

**Comment by nthorne** [comment]
Sounds like good and thorough testing. Do you feel like it is prime for merging in? I've looked through the code and the tests, and you've done a really good job.

**Comment by jbrc** [comment]
Yes, I'm confident in these changes at this point. Missing from this changelist are updates to README.md, but I have those staged in another dev branch and will post a new pull request for those a few hours after this is accepted.

**Comment by nthorne** [comment]
I'll go ahead and merge this one then. By the way - since you've contributed a _lot_ to this one, I think that it is only fair that you are listed as co-copyright holder or at least contributor in the README - which would you prefer?

**Comment by jbrc** [comment]
Thank you for the recognition! I think I would prefer to be listed as a contributor for now. I've been using the add-on (and todo.sh) for only a few months, and I feel like co-copyright holder implies a longer-term involvement with the project. Perhaps if I am still using and helping maintain the add-on a year from now, I would reconsider. Until then, how about a new blurb in the Attribution section of the README? Something like:

Thanks to the following developers for contributing features and fixes:
- jbrc (James Blair)

**Comment by nthorne** [comment]
Done. Thanks for helping out!

**Pull request merged**